### PR TITLE
Fix Publishing Workflows

### PR DESCRIPTION
**Pull request by flavius-t**

### CODE REVIEW

File: .github/workflows/build-and-publish-PyPI.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ name: Build and Publish to PyPI
 
 on:
   push:
-    branches:
-      - 'main'
+    tags:
+      - '*'
 
 jobs:
   build-and-publish:
```
Comment (farzadpanahi) [inline .github/workflows/build-and-publish-PyPI.yml:11]: `python-version: '3.10'` are we using this version in sdk?

File: .github/workflows/build-and-publish-PyPI.yml [hunk 2]
```diff
@@ -28,6 +28,9 @@ jobs:
     - name: Publish Package to PyPI
       uses: pypa/gh-action-pypi-publish@v1.5.0
       with:
-        # TODO: add PyPI user password or access token
         user: __token__
         password: ${{ secrets.PYPI_API_TOKEN }}
+    - name: Release Package
+      uses: ncipollo/release-action@v1.12.0
+      with:
+        commit: main
```
Comment (farzadpanahi) [inline .github/workflows/build-and-publish-PyPI.yml:36]: maybe you don't need this after all

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-publish-PyPI.yml [hunk 3]
```diff
@@ -4,8 +4,8 @@ name: Build and Publish to PyPI
 
 on:
   push:
-    branches:
-      - 'main'
```
[the remark this answers is not in the document]
Comment (flavius-t) [reply .github/workflows/build-and-publish-PyPI.yml:8]: The approach recommended by the maintainer is to use tag pushes as the trigger: https://github.com/ncipollo/release-action

File: .github/workflows/build-and-publish-PyPI.yml (hunk 1 above)
Comment (flavius-t) [reply .github/workflows/build-and-publish-PyPI.yml:11]: Our setup.py specifies python >= 3.6. I used that as a default value, although I am using 3.9 for development. I am not sure which version @sjpark608 is using.

Comment (farzadpanahi) [reply .github/workflows/build-and-publish-PyPI.yml:11]: @sjpark608 how did we come up w/ 3.6?

**Comment by flavius-t** [comment]
TODO:
1. workflow for staging should create tag with name prefix `rc`, i.e. `rc/v.{version-num}` 
2. workflow on push to main, create tag with name prefix `release`, i.e. `release/v.{version-num}`
3. workflow on tag with prefix pattern `release/` should publish to pypi and create release based on the received tag

File: .bumpversion.cfg [hunk 4]
```diff
@@ -1,3 +1,3 @@
 [bumpversion]
-tag = True
-tag_name = "rc/{new_version}"
\ No newline at end of file
+tag_name = "rc/v{new_version}"
+current_version = 1.1.8
```
Comment (farzadpanahi) [inline .bumpversion.cfg:3]: is `current_version = 1.1.8` automatically updated every time we bump version?

**Review by farzadpanahi — APPROVED** [review]

File: .bumpversion.cfg (hunk 4 above)
Comment (flavius-t) [reply .bumpversion.cfg:3]: yes, bumpversion automatically adds/updates this field

File: .github/workflows/build-and-publish-PyPI.yml (hunk 2 above)
Comment (flavius-t) [reply .github/workflows/build-and-publish-PyPI.yml]: i removed it

**Pull request merged**